tests/basic/storage_pool_test.go

```diff
@@ -7923,3 +7923,94 @@ var _ = Describe("{DiffPoolExpansionFromMaintenanceNode}", func() {
 		AfterEachTest(contexts)
 	})
 })
+
+var _ = Describe("{ResyncFailedPoolOutOfRebalance}", func() {
+	// Testrail Description : Resync failed for a volume after pool came out of rebalance PTX-15696 -> PWX-26967
+	/*
+		Deployed systemtest sysbench spec with 1TB volume
+		Pod come up and started writing
+		Added the drive in node 10.13.166.216
+		Observed the volume status to be degraded
+		Waited for pool to come online
+	*/
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("ResyncFailedPoolOutOfRebalance",
+			"Resync failed for a volume after pool came out of rebalance",
+			nil, 0)
+	})
+	var contexts []*scheduler.Context
+	stepLog := "Resync volume after rebalance"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("reducesize-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+
+		// Get Pool with running IO on the cluster
+		poolUUID, err := GetPoolIDWithIOs(contexts)
+		log.FailOnError(err, "Failed to get pool running with IO")
+		log.InfoD("Pool UUID on which IO is running [%s]", poolUUID)
+
+		// Get Node Details of the Pool with IO
+		nodeDetail, err := GetNodeWithGivenPoolID(poolUUID)
+		log.FailOnError(err, "Failed to get Node Details from PoolUUID [%v]", poolUUID)
+		log.InfoD("Pool with UUID [%v] present in Node [%v]", poolUUID, nodeDetail.Name)
+
+		// Resize the Pool few times expanding drives
+
+		poolToBeResized, err := GetStoragePoolByUUID(poolUUID)
+		for count := 0; count < 1; count++ {
+			log.FailOnError(err, "error getting drive size for pool [%s]", poolToBeResized.Uuid)
+			expectedSize := (poolToBeResized.TotalSize / units.GiB) + 50
+
+			// Resize the Pool with either one of the allowed resize type
+
+			log.InfoD("Current Size of the pool %s is %d", poolUUID, poolToBeResized.TotalSize/units.GiB)
+			log.InfoD("Expanding Pool [%v] using resize type [%v]", poolUUID, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK)
+			err = Inst().V.ExpandPool(poolUUID, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize)
+			dash.VerifyFatal(err, nil, "Pool expansion init successful?")
+
+			isjournal, err := isJournalEnabled()
+			log.FailOnError(err, "Failed to check if Journal enabled")
+
+			resizeErr := waitForPoolToBeResized(expectedSize, poolUUID, isjournal)
+			dash.VerifyFatal(resizeErr, nil,
+				fmt.Sprintf("Verify pool %s on expansion using auto option", poolUUID))
+		}
+		log.InfoD("Validate pool rebalance after drive add")
+		err = ValidatePoolRebalance(*nodeDetail, poolToBeResized.ID)
+
+		// Validate Volume resync if any volume got in to resync mode
+		for _, eachContext := range contexts {
+			vols, err := Inst().S.GetVolumes(eachContext)
+			log.FailOnError(err, "Failed to get volumes from context")
```

Comment:
can you add check if vol has already repl3 , if so you can make it 2 and make it 3 again.
Also you can make this operation only on volume which has "poolUUID" as one of its replica

Reply:
done